# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/2d/DrawTargetRecording.cpp
# Commit: 6ea4f3bba40f
# Full Hash: 6ea4f3bba40f2fdf5c4893e0eb76fee7d4e05094
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2025-07-05 09:22:06
# Regressor Bug: 1912897
# File Overlap Count: 1
# Description:
#   Bug 1912897 - Support blurring and shadows for Accelerated Canvas2D. r=aosmond
#   
#   This adds support via FilterNodeWebgl for accelerating specific filters in AC2D.
#   In particular, it adds support for accelerated blurring using a separable blur
#   filter. This is then used to implement the Gaussian Blur filter as well as shadows.
# ==============================================================================

diff -r c6d00b59234d -r 6ea4f3bba40f gfx/2d/DrawTargetRecording.cpp
--- a/gfx/2d/DrawTargetRecording.cpp	Sat Jul 05 00:10:54 2025 +0000
+++ b/gfx/2d/DrawTargetRecording.cpp	Sat Jul 05 04:41:59 2025 +0000
@@ -611,6 +611,22 @@
   return retNode.forget();
 }
 
+already_AddRefed<FilterNode> DrawTargetRecording::DeferFilterInput(
+    const Path* aPath, const Pattern& aPattern, const IntRect& aSourceRect,
+    const IntPoint& aDestOffset, const DrawOptions& aOptions,
+    const StrokeOptions* aStrokeOptions) {
+  RefPtr<FilterNode> retNode = new FilterNodeRecording(mRecorder);
+
+  RefPtr<PathRecording> pathRecording = EnsurePathStored(aPath);
+  EnsurePatternDependenciesStored(aPattern);
+
+  RecordEventSelf(RecordedDeferFilterInput(retNode, pathRecording, aPattern,
+                                           aSourceRect, aDestOffset, aOptions,
+                                           aStrokeOptions));
+
+  return retNode.forget();
+}
+
 void DrawTargetRecording::ClearRect(const Rect& aRect) {
   MarkChanged();
 